parlai/core/teachers.py

```diff
@@ -1257,3 +1261,337 @@ def _setup_data(self, path):
+
+    """
+
+    def __init__(self, opt, shared=None):
+        super().__init__(opt, shared)
+        self.opt = opt
+        self.task = opt['task']
+        self.data_path, self.image_path = self.get_paths(opt)
+
+        self.data = self.load_data(self.data_path, self.opt)
+        self.blank_image_features = torch.FloatTensor(2048).fill_(0)
+        self.datatype = opt.get('datatype').split(':')[0]
+
+        # Example of available models: 'resnet152', 'resnext101_32x48d_wsl',
+        # and ImageLoader supports other resnet and resnext models too
+        self.image_model = opt.get('image_model')
+
+        # Not using default image_mode paramater b/c there is a normalization
+        # (or bug) somewhere in build_dict that is setting it to none
+        self.include_image = opt.get('image_model') != 'no_image_model'
+
+        if shared and 'data' in shared:
+            self.data = shared['data']
+            self.image_loader = shared['image_loader']
+            if 'image_features_dict' in shared:
+                self.image_features_dict = shared['image_features_dict']
+        else:
+            # TODO: Awkward to modify the input opt but needed for the default
+            # TODO: ImageLoader functionality. Is from comment_battle,
+            # TODO: will refactor this at some point soon most likely
+            image_loader_opt = self.opt.copy()
+            image_loader_opt['image_mode'] = (
+                self.image_model if self.include_image else 'none'
+            )
+
+            image_loader_opt['image_size'] = 256
+            image_loader_opt['image_cropsize'] = 224
+            self.image_loader = ImageLoader(image_loader_opt)
+            self.setup_data(self.data_path)
+        self.reset()
+
+    @classmethod
+    def get_available_image_model_names(self):
+        """
+        Available image model names. resnet and resnext variants available
+        from the ImageLoader. resnext101_XXXXX_wsl is the open-sourced FB AI model (960m images, 1.5k hashtags, finetuned on ImageNet).
+
+        """
+        available_model_names = ImageLoader.get_available_model_names()
+        return ['no_image_model'] + available_model_names
+
+    @classmethod
+    def _validate_image_model_name(cls, a):
+        if not isinstance(a, str):
+            raise argparse.ArgumentTypeError(
+                '%s must be a string representing image model name' % a
+            )
+        available_model_names = cls.get_available_image_model_names()
+        if a not in available_model_names:
+            raise argparse.ArgumentTypeError(
+                '%s unknown image model name. Choose from: %s. Currently suggested resnet is resnet152 and resnext is resnext101_32x48d_wsl.'
+                % (a, available_model_names)
+            )
+        return a
+
+    @classmethod
+    def add_cmdline_args(cls, argparser):
+        agent = argparser.add_argument_group('AbstractImageTeacher Arguments')
```

Comment:
This will actually display in the `--help` so a more UI name is probably desirable

Reply:
Hmm - @stephenroller Do you care strongly? I somewhat disagree. If so, I would change it to "Image Teacher Arguments".  @klshuster has otherwise used the exact class name the params set for his CommentBattleModelURU class and I would like to distinguish the teacher params from the ImageLoader params (which are not named for the class and are therefore confusing: "ParlAI Image Preprocessing Arguments").

Reply:
Sure, sounds good.

Reply:
Ok great.